### DISCUSSION TITLE
Content Changelog about last changes across the whole site

### DISCUSSION THREAD

**Discussion by davidhellmann** [craftcms/cms#9103] [Ideas]
We have partly quite complex multisite structures with many authors etc. It is then sometimes difficult to understand what has just happened. 

It would be exciting under Utilities to have a menu item `Content changelog`. where you can see what has happened. And links to these items.

- User X has created / edited Entry XY
- User Y has uploaded / edited Asset XY

Would certainly often help to understand problems. 
Best probably controllable via general.php how long or how many items he saves there between. 

Should be limitable in any case:
- Last 24h
- Last 100 Changes

Opinions on this?
Cheers, David

**Comment by rostockahoi** [comment]
This may be a nice case for a plugin. But it could have some performance implications regarding the amount of data that is generated.

**Reply by davidhellmann** [reply]
I think with limits it's not a problem. Can be cached as other CP stuff and maybe there is a Refresh Button next to it. Maybe it's possible to get this data only if needed. But sure, a Plugin is maybe an approach. Like a query Builder or stuff like that?


Show `all` changes in `all` site(s) from author(s) `all` 
Show `entry` changes in `foo` site(s) from author(s) `max` 
Show `asset` changes in `bar` site(s) from author(s) `claudia`

etc. 

Maybe like the Search It Plugin:
https://plugins.craftcms.com/searchit

**Reply by rostockahoi** [reply]
Could be implemented by hooking into element events for creating/saving/deleting and so on and saving that event information in a common format that fits all element types assets, categories, entries, global sets, matrix blocks?, tags and users.

How could the common format look like?
What about custom element types?
How do entry revivions fit into this?

Displaying them should be straight forward once the common data format is designed properly.

What about diffing content changes, to not make things too easy? :D

**Reply by davidhellmann** [reply]
Diffing is not interesting here I think. Just what parts of the site changed within Period X. Or what did user X the last minutes cause something goes wrong. 

I think it's not necessary to use hooks. I think to make queries via UI is enough here. The query UI should be configurable by a config file: 

```
'elements' => ['entries','assets','products','categories', 'users', 'events']
```
